benchmark/csharp/Tpch/VectorFunctions.cs

```diff
@@ -3,49 +3,70 @@
+            PrimitiveColumn<double> ret = new PrimitiveColumn<double>("Prices", price.Length);
+
+            IEnumerable<ReadOnlyMemory<double>> readOnlyTaxes = tax.GetReadOnlyDataBuffers();
+            IEnumerable<ReadOnlyMemory<double>> readOnlyPrices = price.GetReadOnlyDataBuffers();
+            IEnumerable<ReadOnlyMemory<double>> readOnlyDiscounts = discount.GetReadOnlyDataBuffers();
+
+            IEnumerator<ReadOnlyMemory<double>> taxesEnumerator = readOnlyTaxes.GetEnumerator();
+            IEnumerator<ReadOnlyMemory<double>> pricesEnumerator = readOnlyPrices.GetEnumerator();
+            IEnumerator<ReadOnlyMemory<double>> discountsEnumerator = readOnlyDiscounts.GetEnumerator();
+
+            while (taxesEnumerator.MoveNext() && pricesEnumerator.MoveNext() && discountsEnumerator.MoveNext())
             {
-                builder.Append(prices[i] * (1 - discounts[i]) * (1 + taxes[i]));
+                ReadOnlySpan<double> taxes = taxesEnumerator.Current.Span;
+                ReadOnlySpan<double> prices = pricesEnumerator.Current.Span;
+                ReadOnlySpan<double> discounts = discountsEnumerator.Current.Span;
+                for (int i = 0; i < prices.Length; ++i)
+                {
+                    ret[i] = (prices[i] * (1 - discounts[i]) * (1 + taxes[i]));
+                }
             }
 
-            return builder.Build();
+            return ret;
         }
 
-        internal static DoubleArray ComputeDiscountPrice(DoubleArray price, DoubleArray discount)
+        internal static PrimitiveColumn<double> ComputeDiscountPrice(PrimitiveColumn<double> price, PrimitiveColumn<double> discount)
         {
             if (price.Length != discount.Length)
             {
                 throw new ArgumentException("Arrays need to be the same length");
             }
 
-            int length = price.Length;
-            var builder = new DoubleArray.Builder().Reserve(length);
-            ReadOnlySpan<double> prices = price.Values;
-            ReadOnlySpan<double> discounts = discount.Values;
-            for (int i = 0; i < length; ++i)
+            PrimitiveColumn<double> ret = new PrimitiveColumn<double>("Prices", price.Length);
+
+            IEnumerable<ReadOnlyMemory<double>> readOnlyPrices = price.GetReadOnlyDataBuffers();
+            IEnumerable<ReadOnlyMemory<double>> readOnlyDiscounts = discount.GetReadOnlyDataBuffers();
+
+            IEnumerator<ReadOnlyMemory<double>> pricesEnumerator = readOnlyPrices.GetEnumerator();
+            IEnumerator<ReadOnlyMemory<double>> discountsEnumerator = readOnlyDiscounts.GetEnumerator();
+
+            while (pricesEnumerator.MoveNext() && discountsEnumerator.MoveNext())
             {
-                builder.Append(prices[i] * (1 - discounts[i]));
+                ReadOnlySpan<double> prices = pricesEnumerator.Current.Span;
+                ReadOnlySpan<double> discounts = discountsEnumerator.Current.Span;
+                for (int i = 0; i < prices.Length; ++i)
```

Comment:
Is it possible to do this one loop?
Ideally, 
```
while (prices.MoveNext() && discounts.MoveNext())
{
    ret[i] = (prices.Current() * (1 - discounts.Current()));
}
```
It may be less performant b/c of all the function calls, but it will be good to have for the user experience?

The most ideal user experience will be:
```C#
var ret = prices * (1 - discounts);
```
similar to Pandas. I am curious whether this can be done. :)
```